### PR TITLE
Unit test cases for SpyMemcached

### PR DESCRIPTION
**Pull request by IshikaDawda**
### Merge After [PR#112](https://github.com/newrelic/csec-java-agent/pull/112)
Unit test cases for SpyMemcached module.

### CODE REVIEW

File: instrumentation-security/spymemcached-2.12.0/src/test/java/com/nr/agent/security/instrumentation/memcached/test/MemcachedTest.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    private final String key = "someKey";
+    private final String value = "value";
+    private final int expirationInSeconds = 1800;
+    private final long casID = 1;
+    private static JMemcachedServer server;
+
+    @BeforeClass
+    public static void setup() throws IOException {
+        server = new JMemcachedServer();
+        server.start("127.0.0.1", 11214);
```
Comment (monu-k2io) [inline instrumentation-security/spymemcached-2.12.0/src/test/java/com/nr/agent/security/instrumentation/memcached/test/MemcachedTest.java:38]: do not hardcode the port, instead get the random port and use that

File: instrumentation-security/spymemcached-2.12.0/src/test/java/com/nr/agent/security/instrumentation/memcached/test/MemcachedTest.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    }
+
+    @AfterClass
+    public static void stop() {
+        memcachedClient.flush();
+        memcachedClient.shutdown();
+        server.clean();
+    }
+    @Test
+    public void setTest() {
```
Comment (monu-k2io) [inline instrumentation-security/spymemcached-2.12.0/src/test/java/com/nr/agent/security/instrumentation/memcached/test/MemcachedTest.java:49]: follow the naming convention that we have been using for the method name `testSet()`

**Review by monu-k2io — CHANGES_REQUESTED** [review]

File: instrumentation-security/spymemcached-2.12.0/src/test/java/com/nr/agent/security/instrumentation/memcached/test/MemcachedTest.java (hunk 1 above)
Comment (IshikaDawda) [reply instrumentation-security/spymemcached-2.12.0/src/test/java/com/nr/agent/security/instrumentation/memcached/test/MemcachedTest.java]: Updated in this [commit](https://github.com/newrelic/csec-java-agent/pull/111/commits/cc2abafee8e14573711eb65bb718458d1efc55ef)

File: instrumentation-security/spymemcached-2.12.0/src/test/java/com/nr/agent/security/instrumentation/memcached/test/MemcachedTest.java (hunk 2 above)
Comment (IshikaDawda) [reply instrumentation-security/spymemcached-2.12.0/src/test/java/com/nr/agent/security/instrumentation/memcached/test/MemcachedTest.java]: Updated in this [commit](https://github.com/newrelic/csec-java-agent/pull/111/commits/cc2abafee8e14573711eb65bb718458d1efc55ef)

**Review by lovesh-ap — APPROVED** [review]

**Pull request merged**